op-supervisor/supervisor/backend/backend.go

```diff
@@ -159,6 +159,46 @@ func (su *SupervisorBackend) CheckMessage(identifier types.Identifier, payloadHa
 	return safest, nil
 }
 
+func (su *SupervisorBackend) CheckMessages(
+	identifiers []types.Identifier,
+	payloadHashes []common.Hash,
+	minSafety types.SafetyLevel) error {
+	if len(identifiers) != len(payloadHashes) {
+		return errors.New("identifiers and payload hashes must be the same length")
+	}
+	for i, identifier := range identifiers {
+		safety, err := su.CheckMessage(identifier, payloadHashes[i])
+		if err != nil {
+			return fmt.Errorf("failed to check message: %w", err)
+		}
+		if !atLeastAsSafe(safety, minSafety) {
+			return fmt.Errorf("message %v is not safe: %v", identifier, safety)
+		}
+	}
+	return nil
+}
+
+// TODO: Really this should be a method on types.SafetyLevel
+// for now it's easy to do this calculation here
+func atLeastAsSafe(safety, minSafety types.SafetyLevel) bool {
```

Comment:
I will make this a function of `types.SafetyLevel` tomorrow, as it's just as easy to have it embedded over there than left here.